query_plan.go

```diff
@@ -143,7 +151,9 @@ func (n *Node) RenderTreeWithStats(planNodes []*pb.PlanNode) ([]QueryPlanRow, er
 		var predicates []string
 		for _, cl := range planNodes[planNode.ID].GetChildLinks() {
 			child := planNodes[cl.ChildIndex]
-			if child.DisplayName != "Function" || !(cl.GetType() == "Residual Condition" || cl.GetType() == "Seek Condition" || cl.GetType() == "Split Range") {
+			// Known predicates are Condition(Filter) or Seek Condition/Residual Condition(FilterScan) or Split Range(Distributed Union).
+			// Agg is a Function but not a predicate.
+			if child.DisplayName != "Function" || (!strings.HasSuffix(cl.GetType(), "Condition") && cl.GetType() != "Split Range") {
```

Comment:
By the way, what is the reason to have condition `child.DisplayName != "Function"`? If I understand correctly, `Function` operator is not visible in our query plan.

Reply:
Predicates are subset of `Function` operators which returns bool so `Function` is not visible as a node of the operator tree but needs to be visible as a predicate.
Although, there are no return type information in `PlanNode` so it is whitelisted by the type name of `ChildLink` which link to the `Function` operator.

Currently, I don't yet see a child link like which type is `*Condition` and child is not `Function` so only `ChildLink.type` is needed information to distinguish predicates but I think that to be `Function` is a necessary condition to be a predicate so I require it explicitly.

Reply:
Extract isPredicate function in https://github.com/cloudspannerecosystem/spanner-cli/pull/87/commits/876f8382a762fb6d026afabc8e4f02b9db854ba3

Reply:
Ah that makes sense. Thank you for clarifying it!